File: catalog/backends/java-spring/patterns/SsrfExampleService.java
package com.example.service;

import org.springframework.web.bind.annotation.*;
import org.springframework.web.client.RestTemplate;

// Pattern: Code Security — SSRF via unvalidated URL parameter
// WARNING: intentionally vulnerable for IAST demo
// Adapt: use a domain-appropriate endpoint name
@RestController
public class SsrfExampleService {

    @GetMapping("/api/fetch-url")
    public String fetchUrl(@RequestParam String url) {
        return new RestTemplate().getForObject(url, String.class);
    }
}

File: catalog/backends/dotnet-aspnetcore/patterns/SsrfExampleService.cs
// Pattern: AppSec SSRF — unvalidated URL passed to HttpClient
// Adapt: wrap in a domain-specific proxy or fetcher endpoint
public static class SsrfExampleService
{
    private static readonly HttpClient Client = new();

    public static void Map(WebApplication app)
    {
        app.MapGet("/fetch", async (string url) =>
        {
            // INTENTIONALLY VULNERABLE — demonstrates SSRF for Datadog Code Security
            var body = await Client.GetStringAsync(url);
            return Results.Text(body, "text/plain");
        });
    }
}
